core/managers.py

```diff
@@ -41,6 +41,15 @@ def agents_only(self):
     def structures_only(self):
         return self.exclude(structure__isnull=True)
 
+    def with_active_user(self):
+        return self.filter(Q(structure__isnull=False) | Q(agent__user__is_active=True))
+
+    def exclude_empty_emails(self):
+        return self.exclude(email="")
+
+    def can_be_emailed(self):
+        return self.exclude_empty_emails().with_active_user()
+
```

Comment:
Je trouve plus claire de l'écrire tel quel, qu'en penses-tu ? 
```suggestion
    def can_be_emailed(self):
        """
        Retourne les contacts qui peuvent recevoir des emails selon les règles :
        - Pour un contact d'agent : email renseigné + utilisateur actif
        - Pour un contact de structure : email renseigné uniquement
        """
        agent_contacts = self.exclude_empty_emails().with_active_agent()
        structure_contacts = self.exclude_empty_emails().with_structure_only()
        
        return agent_contacts | structure_contacts
```